src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8517,40 +8509,168 @@ protected MethodGroupResolution ResolveExtension(
+                LookupOptions lookupOptions,
+                Binder binder,
+                ExtensionScope scope,
+                BindingDiagnosticBag diagnostics,
+                out MethodGroupResolution result)
+            {
+                Debug.Assert(left.Type is not null);
+                result = default;
+
+                // 1. gather candidates
+                scope.Binder.LookupExtensionMembersInSingleBinder(
+                    lookupResult, left.Type, memberName, arity,
+                    basesBeingResolved: null, lookupOptions, originalBinder: binder, ref useSiteInfo);
+
+                CompoundUseSiteInfo<AssemblySymbol> classicExtensionUseSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics);
+                binder.LookupExtensionMethods(classicExtensionLookupResult, scope, memberName, arity, ref classicExtensionUseSiteInfo);
                 diagnostics.Add(expression, classicExtensionUseSiteInfo);
 
                 lookupResult.MergeEqual(classicExtensionLookupResult);
 
                 if (!lookupResult.IsMultiViable)
                 {
-                    diagnostics.Free();
-                    continue;
+                    return false;
                 }
 
-                var members = ArrayBuilder<Symbol>.GetInstance();
-                Symbol? symbol = this.GetSymbolOrMethodOrPropertyGroupStrict(lookupResult, expression, memberName, arity, members, diagnostics, qualifierOpt: null);
-                if (symbol is not null)
+                // 2. type inference, applicability to arguments and overload resolution for methods
+                MethodGroupResolution methodResult = analyzeMethods(lookupResult, binder, left, typeArgumentsWithAnnotations, options, analyzedArguments, ref actualArguments, diagnostics, returnRefKind, returnType, expression, in callingConvention);
+
+                // 3. type inference, applicability to receiver for properties
+                ArrayBuilder<PropertySymbol>? applicableProperties = analyzeProperties(lookupResult, binder, left, ref useSiteInfo, out bool foundProperties);
+
+                // 4. determine member kind
+                if (!methodResult.HasAnyApplicableMethod && applicableProperties is null)
                 {
-                    firstResult.Free();
-                    actualArguments?.Free();
-                    lookupResult.Free();
-                    classicExtensionLookupResult.Free();
-                    members.Free();
-                    return new MethodGroupResolution(symbol, LookupResultKind.Viable, diagnostics.ToReadOnlyAndFree());
+                    // Found nothing applicable. Store aside the first non-applicable result and continue searching for an applicable result.
+                    if (firstResult.IsEmpty)
+                    {
+                        firstResult = foundProperties ? makeErrorResult(binder, left.Type, memberName, arity, lookupResult, expression, diagnostics) : methodResult;
```

Comment:
:memo: Note that when not properties are found, we use the `methodResult` which contains the overload resolution results and therefore can describe what went wrong in details (`OverloadResolutionResult.ReportDiagnostics`). This ensures we don't regress existing scenarios. But when properties are involved we only report a single/generic diagnostic at the moment.